app/experimenter/nimbus-ui/webpack.config.js

```diff
@@ -0,0 +1,21 @@
+const path = require("path");
+const glob = require("glob");
+const UglifyJsPlugin = require("uglifyjs-webpack-plugin");
+
+// NOTICE: This is a suplementary Webpack config used for production
+// building our JavaScript, and is not the config create-react-app
+// uses for running and building the React app.
```

Comment:
This file is missing the license.

We could name this `webpack.config.prod.js` which is conventional when there's a dev/prod version of webpack but I think it's also fine to leave with the comment. It could also be useful to add _why_ we need this like "we conditionally require JS files in `nimbus-ui.html` which enables us to pass server configuration to the client and reduce them to one here" for extra breadcrumbs but not necessary.

This sorta makes me wonder if we need to regenerate and also expose source maps though? I think Sentry uses those for error tracebacks. 😓

I guess bonus points would be removing all those files since there's now extra cruft in that `js` folder inside the build that won't be used, but meh. Something like this could be a good gulp/grunt task, but we already have webpack as a dependency, so hey less the better.

Reply:
File name updated, note added, license added, leftover build files removed after joining. Unsure about source maps right now, but there's one issue I'm stuck on:

I can't get this to actually work. I am joining the CSS/JS files in the same order CRA places them if they were individually rendered in the HTML, they're not being additionally minified, the JS file loads on the page, and there are no console errors, but in testing the React app doesn't start (blank page). My theory at the moment is that Webpack is adding some additional bundling mechanics to the mix that is making the React script not work. But I'm officially looking for input if anyone has any (cc @jaredlockhart and @lmorchard).

(Context for others: in production I'm using [Webpack to bundle](https://github.com/mozilla/experimenter/pull/3670/files#diff-ff3e1bc0781aa8c83a318f2e5b9a3315) all of CRA's output files in to a single `bundle.min.js` so it can be rendered in [the template](https://github.com/mozilla/experimenter/pull/3670/files#diff-23698a628d30539e82056573789ef0eaR27).)

Reply:
FWIW this all works as expected in development, so if we want to merge this to unblock other stuff I can follow up.

Reply:
This is beyond my limited knowledge 😞

Reply:
Cancel this. Taking a different approach.